pkg/istioversions/versions.yaml

```diff
@@ -11,6 +11,8 @@
 # go.mod affect the generated API schema for the Sail CRDs (e.g. IstioRevision),
 # as well as all the Istio CRDs (e.g. VirtualService).
 versions:
+  - name: v1.24-latest
+    version: 1.24.2
```

Comment:
Hmm, this is now confusing. The `version` field now has two different purposes. In an alias entry, it points to the `name` field of another entry in the list, but in an actual version entry, it specifies the version number. I think we should use a different name as we did before (`ref` or `targetName` or `target` or `resolvesTo` or `aliasFor` or something similar).

Reply:
I actually want to avoid introducing a new field on the struct. 

- For the `versions` array, adding `aliasFor` will mean introducing a new element type in the array. Is this a good idea? 

- If both `version` and `aliasFor` are specified in an item, should we fail early or prioritize `aliasFor` over the other and warn user?